### DISCUSSION TITLE
Link names to events

### DISCUSSION THREAD

**Discussion by fisharebest** [fisharebest/gedcom-name#2] [General]
Many names are linked to events where they are created.  For example:

* adoption - adopted name
* marriage - married name
* confirmation - religious name
* immigration - imigrant name

Do we need the ability to link names to events?

**Comment by hartenthaler** [comment]
Your examples in the topic above are all related to events in the life of a person. But there is another type of event that could change a name. For example, in the Austria-Hungary monarchy, it happened that names of persons are changed from a former Hungary name to a more German version of the name, maybe you can compare it to what happened sometimes at an immigration event. In this case, it happened even if the person does not change the living location.

I just searched for a source but didn't find it. So maybe somebody else could add a source for such a name-changing event.

**Comment by hartenthaler** [comment]
> Do we need the ability to link names to events?

Yes, that would be great. Otherwise, you can only establish a relation by comparing the date values in the NAME and in the event subtag (marriage, adoption, ...).

**Comment by hartenthaler** [comment]
Link names to events? Link events to names? Or both?
